src/main/scala/cognite/spark/v1/NumericDataPointsRdd.scala

```diff
@@ -249,7 +249,10 @@ final case class NumericDataPointsRdd(
         queryDatapointsById(id, start, end.max(start.plusMillis(1)), 1)
           .map(datapoints => id -> datapoints.headOption)
       }.parSequence
-      latest <- client.dataPoints.getLatestDataPoints(ids, ignoreUnknownIds = true)
+      latest <- client.dataPoints.getLatestDataPoints(
+        ids,
+        ignoreUnknownIds = true,
+        end.toEpochMilli.toString)
```

Comment:
how does this change help with this issue? could you explain 😅  is the default value now here and do we override it?

Reply:
instead of using `"now"`, which is the API default, we will use our upper bound. If user supplied a filter `timestamp < X', then we'll use X. If not, we have some hardcoded date at 2100. I hope nobody is trying to predict valve flow at year 2100 :sweat_smile:

Reply:
actually I encountered an error that timestamp can not be greater than the year 2050 when trying to insert some dataPoints to test, should be fine I think 😁

Reply:
BAN  ‼️ 

![image](https://user-images.githubusercontent.com/58691727/140524239-5405bc88-8b7a-4a9d-94bc-8c1d08066e5d.png)

Reply:
Aha yes, I didn't see that fix, thanks!

Reply:
Yes, but the git commit history nor the source code will record this, so a comment in either the commit message or the code itself about why this change is done, would have been useful.

Reply:
done ;) https://github.com/cognitedata/cdp-spark-datasource/pull/536/commits/e4eedd3adbaf73cc0a8dedd73ebd09f7473e76b3